src/boundaries.rs

```diff
@@ -163,48 +163,69 @@ pub fn build_boundary_parts<T: Borrow<osmpbfreader::OsmObj>>(
     // we want to try to build a polygon for a least each way
     while !boundary_parts.is_empty() {
         let first_part = boundary_parts.remove(0);
-        let first = first_part.first();
+        let mut added_nodes: Vec<osmpbfreader::Node> = vec![];
+        let mut node_to_idx: BTreeMap<osmpbfreader::NodeId, usize> = BTreeMap::new();
+
+        let mut append_ring = |nodes: &[osmpbfreader::Node]| {
+            let poly_geom = nodes
+                .iter()
+                .map(|n| Coordinate {
+                    x: n.lon(),
+                    y: n.lat(),
+                })
+                .collect();
+            multipoly
+                .0
+                .push(Polygon::new(LineString(poly_geom), vec![]));
+        };
```

Comment:
It would probably be clearer to define this helper function outside of the while loop, probably right next the definition of `multipoly` as this is the only context it depends on?

Reply:
done